sdk/servicebus/service-bus/src/util/utils.ts

```diff
@@ -522,6 +522,7 @@ export async function waitForTimeoutOrAbortOrResolve<T>(args: {
     }
   };
 
+  // eslint-disable-next-line promise/param-names
```

Comment:
A few lines below, you have 
```js
const actionPromise = args.actionFn();
try {
    return await Promise.race([abortOrTimeoutPromise, actionPromise]);
} finally {
    clearAbortSignalAndTimer();
}
```

The call to `args.actionFn()` should be inside the try block. Otherwise when `actionfn()` fails, we fail to clear the timer and the abort event listener

We can remove the constant and call `args.actionFn()` directly inline

Reply:
Moving it in seems very sensible. Will do.

Reply:
Done. 

Looking at it now I think the surface area for it to throw was small (it has to throw before the promise is returned) but there's zero reason for me to not handle that properly. :)